sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1607,7 +1607,9 @@ class Dataset[T] private[sql](
    */
   @Experimental
   @InterfaceStability.Evolving
-  def reduce(func: (T, T) => T): T = rdd.reduce(func)
+  def reduce(func: (T, T) => T): T = withNewExecutionId {
```

Comment:
Only a place to need this wrapper in this file?

@gatorsmile btw, this api is still useful? Or, this intends to be deprecated in v3.0+?

Reply:
cc @zsxwing

Reply:
@maropu When you asked about this API did you refer to `reduce` or `withNewExecutionId`?

Reply:
`reduce`

Reply:
Why would we want to deprecate it?

Reply:
@rxin Sorry, I missed that comment. I just thought what `reduce` can do is what type-safe `groupByKey` can do. But, probably I missed something.